crates/swc_ecma_transforms_module/src/system_js.rs

```diff
@@ -0,0 +1,1143 @@
+use crate::{
+    path::{ImportResolver, NoopImportResolver},
+    util::{local_name_for_src, use_strict},
+};
+use anyhow::Context;
+use serde::{Deserialize, Serialize};
+use swc_atoms::{js_word, JsWord};
+use swc_common::{collections::AHashMap, FileName, Mark, Span, DUMMY_SP};
+use swc_ecma_ast::*;
+use swc_ecma_utils::{
+    ident::IdentLike, member_expr, private_ident, quote_ident, quote_str, undefined,
+    var::VarCollector, ExprFactory,
+};
+use swc_ecma_visit::{noop_fold_type, Fold, FoldWith, VisitWith};
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+#[serde(deny_unknown_fields, rename_all = "camelCase")]
+pub struct Config {
+    #[serde(default)]
+    pub allow_top_level_this: bool,
+}
+
+struct SystemJs<R>
+where
+    R: ImportResolver,
+{
+    root_mark: Mark,
+    resolver: Option<(R, FileName)>,
+    config: Config,
+
+    declare_var_idents: Vec<Ident>,
+    export_map: AHashMap<Id, Vec<JsWord>>,
+    export_names: Vec<JsWord>,
+    export_values: Vec<Expr>,
+    tla: bool,
+    enter_async_fn: u32,
+    enter_fn: u32,
+    root_fn_decl_idents: Vec<Ident>,
+    module_item_meta_list: Vec<ModuleItemMeta>,
+    import_idents: Vec<Id>,
+    export_ident: Ident,
+    context_ident: Ident,
+}
+
+pub fn system_js(root_mark: Mark, config: Config) -> impl Fold {
+    SystemJs {
+        root_mark,
+        resolver: None::<(NoopImportResolver, _)>,
+        config,
+
+        declare_var_idents: vec![],
+        export_map: Default::default(),
+        export_names: vec![],
+        export_values: vec![],
+        tla: false,
+        enter_async_fn: 0,
+        enter_fn: 0,
+        root_fn_decl_idents: vec![],
+        module_item_meta_list: vec![],
+        import_idents: vec![],
+        export_ident: private_ident!("_export"),
+        context_ident: private_ident!("_context"),
+    }
+}
+
+pub fn system_js_with_resolver<R>(
+    resolver: R,
+    base: FileName,
+    root_mark: Mark,
+    config: Config,
+) -> impl Fold
+where
+    R: ImportResolver,
+{
+    SystemJs {
+        root_mark,
+        resolver: Some((resolver, base)),
```

Comment:
Can you change this to use https://github.com/swc-project/swc/blob/ded7a8e0a96842e48a4aea7cca565b4c22b45da8/crates/swc_ecma_transforms_module/src/path.rs#L16-L22 instead?

This is important for reducing size of the binary.